Fix out-of-bounds write

gaoethan opened the pull request:
The following code merely checks length greater than 200:

`if (length > BUFFERLENGTH)`

that means the `length` may equal to 200. so it will overrun
array buf of 200 bytes at byte offset 200 for the following
code when `length` is 200:

```
memcpy(buf, fmt, length);
buf[length] = 0;
```

Signed-off-by: Ethan Gao <ethan.gao@linux.intel.com>

mikaelarguedas commented:
Thanks @gaoethan,
This being a fork of uncrustify we try to diverge as less as possible to ease syncing with upstream in the future, so I'm going to close this. Please consider making the contribution [upstream](https://github.com/uncrustify/uncrustify)

gaoethan committed:
* Fix out-of-bounds write
The following code merely checks length greater than 200:

if (length > BUFFERLENGTH)

that means the length may equal to 200. so it will overrun
array buf of 200 bytes at byte offset 200 for the following
code when length is 200:

memcpy(buf, fmt, length);
buf[length] = 0;

Signed-off-by: Ethan Gao <ethan.gao@linux.intel.com>

gaoethan commented:
that's okay. let me submit it directly to upstream uncrustify
